parser/prism/Translator.cc

```diff
@@ -30,12 +30,24 @@ std::unique_ptr<SorbetAssignmentNode> Translator::translateAssignment(pm_node_t
 
     auto node = reinterpret_cast<PrismAssignmentNode *>(untypedNode);
     auto *loc = &node->base.location;
-    auto *nameLoc = &node->name_loc;
 
-    auto name = parser.resolveConstant(node->name);
-    auto lhs = make_unique<SorbetLHSNode>(parser.translateLocation(nameLoc), gs.enterNameUTF8(name));
+    unique_ptr<parser::Node> lhs;
     auto rhs = translate(node->value);
 
+    if constexpr (std::is_same_v<PrismAssignmentNode, pm_index_operator_write_node> ||
+                  std::is_same_v<PrismAssignmentNode, pm_index_and_write_node> ||
+                  std::is_same_v<PrismAssignmentNode, pm_index_or_write_node>) {
+        auto *openingLoc = &node->opening_loc;
+        auto receiver = translate(node->receiver);
+        auto args = translateArguments(node->arguments);
+        lhs = make_unique<parser::Send>(parser.translateLocation(loc), std::move(receiver), gs.enterNameUTF8("[]"),
```

Comment:
I think this can be interned.

```suggestion
        lhs = make_unique<parser::Send>(parser.translateLocation(loc), std::move(receiver), [gs.enterNameUTF8("[]")](core::Names::squareBrackets()),
```

Compare:

https://github.com/Shopify/sorbet/blob/9fa4e01b8fe3dfb481976cc28f28a16909578830/parser/Builder.cc#L1032-L1036